### PR TITLE
[API compatibility] add paddle nn.functional.dropout1d api

### PR DESCRIPTION
**Pull request by zhanghonggeng**
### PR Category
User Experience

### PR Types
New features

### Description
增加paddle nn.functional.dropout1d(input, p=0.5, training=True, inplace=False) api
参数:
        input (Tensor): 输入张量，形状需为 [C, L] 或 [N, C, L]
        p (float): 置零概率（0-1），默认 0.5
        training (bool): 是否训练模式，默认 True
        inplace (bool): Paddle 暂不支持，仅为兼容 API
WARNING: 
当前dropout,dropout1d,dropout2d,dropout3d均不支持inpalce，后续会在dropout中实现dropout_nd C++下沉，同时支持inpacle。
Note：
在 paddle.nn.functional.dropout 中，axis 参数控制 dropout 的应用维度：
当 axis=None（默认值）：对输入 Tensor x 中的任意元素，以丢弃概率 p 随机将一些元素输出置 0。
当 axis≠None 时（如 axis=1）：以一定的概率从图像特征或语音序列中丢弃掉整个通道。

虽然文档中建议 axis 保持默认值，但这里将 axis 设置为 1 更符合 dropout1d 的含义。
image

pcard-67164

### CODE REVIEW

File: python/paddle/nn/functional/common.py [hunk 1]
```diff
@@ -1427,6 +1427,66 @@ def get_attrs(prog, dropout_prob, is_test, seed):
...
+    inplace: bool = False,
+) -> paddle.Tensor:
+    """
+    Randomly zero out entire 1D channels (feature maps) during training.
+
+    Args:
+        input: Input tensor of shape [C, L] (2D) or [N, C, L] (3D)
+        p: Probability of a channel being zeroed. Default: 0.5
+        training: If False, returns input unchanged. Default: True
+        inplace: Not supported in Paddle (for API compatibility only)
```
Comment (zhwesky2010) [inline python/paddle/nn/functional/common.py]: 这个应该可以实现inplace，ymal中能加inplace吗

**Review by zhwesky2010 — COMMENTED** [review]
看看inplace这个问题，目前yaml里支持了inplace，能否支持一个inplace版本的dropout_接口。inpplace=False时调用dropout，inplace=True时，调用dropout_。

或者通过assign(output)这种能否支持？

File: python/paddle/nn/functional/common.py (hunk 1 above)
Comment (zhanghonggeng) [reply python/paddle/nn/functional/common.py]: 可以通过assign的方式实现inplace。当前 dropout 的实现中当axis≠None时是手动计算 mask 和缩放，不调用 _C_ops.dropout ，增加 dropout__ 也不能覆盖这条路径。

File: test/legacy_test/test_dropout_op.py [hunk 2]
```diff
@@ -919,6 +919,105 @@ def test_dygraph(self):
...
+            )
+            in_np = np.random.random(input_shape).astype("float32")
+            exe = base.Executor(place)
+            fetches = exe.run(
+                main_prog,
+                feed={input_name: in_np},
+                fetch_list=[res],
+            )
+
+            np.testing.assert_allclose(fetches[0], in_np, rtol=1e-05)
```
Comment (zhwesky2010) [inline test/legacy_test/test_dropout_op.py:948]: rtol应该不用降低

File: test/legacy_test/test_dropout_op.py [hunk 3]
```diff
@@ -919,6 +919,105 @@ def test_dygraph(self):
...
+                training=False,
+                p=0.0,
+            )
+
+            self.check_static_result(
+                place=place,
+                input_name="input_3d",
+                input_shape=[2, 3, 4],
+                training=False,
+                p=0.0,
```
Comment (zhwesky2010) [inline test/legacy_test/test_dropout_op.py:965]: p=1.0 测一下

**Review by zhwesky2010 — COMMENTED** [review]
assing实现的话可能得充分测下反向，看看与竞品的inplace=True下反向是不是一样

**Comment by zhwesky2010** [comment]
dropout=0下rtol是不是不用降低

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
